### PR TITLE
Created method for viewing pictures in large mode

### PR DESCRIPTION
**Pull request by teamagam5**
Created method for viewing pictures in large mode

### CODE REVIEW

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadPictureTask.java [hunk 1]
```diff
@@ -10,19 +10,23 @@
...
     }
 
     @Override
     protected Bitmap doInBackground(String... params) {
         String path = params[0];
-        return decodeSampledBitmapFromFile(path, mImageSize, mImageSize);
+        return BitmapFactory.decodeFile(path);
     }
 
     @Override
```
Comment (BarAgam) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadPictureTask.java:32]: I cannot comment on line 36, but it seems like you're checking the weak reference for null value twice.

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadPictureTask.java [hunk 2]
```diff
@@ -10,19 +10,23 @@
...
 
-    LoadPictureTask(ImageView imageView, int imageSize) {
+    protected final WeakReference<ImageView> mImageViewReference;
+
+    public LoadPictureTask() {
+        super();
+        mImageViewReference = null;
+    }
+
+    LoadPictureTask(ImageView imageView) {
```
Comment (BarAgam) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadPictureTask.java:20]: Should you call super() here aswell?

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import android.view.View;
+import android.widget.ImageView;
+
+import java.lang.ref.WeakReference;
+
+class LoadThumbnailTask extends LoadPictureTask {
+    private final int mImageSize;
+
+    LoadThumbnailTask(ImageView imageView, int imageSize) {
+        // Use a WeakReference to ensure the ImageView can be garbage collected
```
Comment (BarAgam) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java]: You're not using a weak reference here. comment is outdated. (copy-paste?)

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Override
+    protected Bitmap doInBackground(String... params) {
+        String path = params[0];
+        return decodeSampledBitmapFromFile(path, mImageSize, mImageSize);
+    }
+
+    private static Bitmap decodeSampledBitmapFromFile(String path, int requiredWidth, int requiredHeight) {
+
+        // First decode with inJustDecodeBounds=true to check dimensions
```
Comment (BarAgam) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java]: This method is a somewhat long.
Each comment here can be replaced with extracted the following code block into a new method with appropriate name.

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        // Calculate inSampleSize
+        options.inSampleSize = calculateSampleSize(options, requiredWidth, requiredHeight);
+
+        // Decode bitmap with inSampleSize set
+        options.inJustDecodeBounds = false;
+        return BitmapFactory.decodeFile(path, options);
+    }
+
+    private static int calculateSampleSize(Options options, int requiredWidth, int requiredHeight) {
+        // Raw height and width of image
```
Comment (BarAgam) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java:40]: Same.
Small methods => profit.

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java [hunk 6]
```diff
@@ -135,4 +137,12 @@ private PictureInfo pictureInfoFromFile(File file) {
         return pictureInfo;
     }
 
+    private void enlargePicture(String path) {
+        final Dialog enlargePictureDialog = new Dialog(this);
```
Comment (BarAgam) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/MainActivity.java]: I'd create a class that extends dialog that handles all that is necessary for the image displaying.
You could also add some static method to it that initializes a new Dialog given a path, shows it, and execute it's appropriate async-task.

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/PictureAdapter.java [hunk 7]
```diff
@@ -34,12 +34,12 @@ public void onBindViewHolder(PictureViewHolder pictureViewHolder, int position)
                 .getApplicationContext()
                 .getResources()
                 .getInteger(R.integer.activity_main_thumbnail_size);
-        if (null != pictureViewHolder.mLoadPictureTask) {
-            pictureViewHolder.mLoadPictureTask.cancel(true);
+        if (null != pictureViewHolder.mLoadThumbnailTask) {
```
Comment (BarAgam) [inline DailySelfie/app/src/main/java/com/teamagam/dailyselfie/PictureAdapter.java]: I'd refactor this method into smaller ones.

**Review by BarAgam — APPROVED** [review]

File: DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java (hunk 5 above)
Comment (teamagam5) [reply DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java:40]: It was taken from developer.android.com & I didn't find a proper way to divide it so as for now I leave it as is

**Pull request merged**